src/main/java/entities/User.java

```diff
@@ -0,0 +1,170 @@
+package java.entities;
+import java.util.ArrayList;
+
+public class User {
+    // ArrayList of User's favourited posts
+    private final ArrayList<Post> favourites;
+
+    // ArrayList of Posts User has made
+    private final ArrayList<Post> posts;
+
+    // boolean relating if User has admin privileges
+    private final boolean isAdmin;
+
+    // Number of Users created
+    public static int numUsers = 0;
+
+    // id is the unique identifier of User, equal to number of posts at time of creation
+    private final int id;
+
+    // password is the user's input password, used to sign in
+    private final String password;
+
+    // email is the users input email, used to sign in
+    private final String email;
+
+    /*
+    * Initializes an instance of User
+    *
+    * @param admin whether the user will have admin priviledges
+    * */
+    public User(boolean admin, int numUsersCreated, String email, String password) {
+        this.favourites = new ArrayList<Post>();
+        this.posts = new ArrayList<Post>();
+        this.isAdmin = admin;
+        User.numUsers = numUsersCreated;
```

Comment:
Since it's static and we increment it each time a user is created, there's no need to have it as a parameter or re-assign numUsers. Just having the increment should be fine.

Reply:
We discussed this and realized that the class variable doesn't persist in the database, so we need this feature to keep a proper id system.